source/commands/command_factory.py

```diff
@@ -0,0 +1,24 @@
+import os,sys
+sys.path.append(os.path.join(os.getcwd(), "commands"))
+
+from commands.create_command import CreateCommand
+from source.output.exceptions import *
+import os,sys
+
+class CommandFactory:
+    def __init__(self, args):
+        self.args = args
+        self.available_commands = {"create", "set", "get", "delete"}
+        self.validate(self)
+
+    @staticmethod
+    def validate(self):
+        if self.args.command.lower() not in self.available_commands:
```

Comment:
an error would be raised if command is None because you can't use lower() with None. you need to check if it is none or not first.

Reply:
Ok I will edit this. Thanks